rust/src/lib.rs

```diff
@@ -252,6 +256,11 @@ pub mod ffi {
     // lockfile.rs
     extern "Rust" {
         fn ror_lockfile_read(filenames: &Vec<String>) -> Result<Vec<StringMapping>>;
+        fn ror_lockfile_write(
+            filename: &str,
+            packages: Vec<u64>,
```

Comment:
My biggest concern here is basically that I want to be able to grep for this so we can find the places that do this more easily and fix them later, and `u64` is not a grep friendly technique.

I think one thing that might work is to generically bind `GObject*` - then we just do casts on both sides?

Reply:
I'm going to play with this a bit.

Reply:
Good point re. grepping. Maybe we can add markers like `// HACK: u64 ptr: DnfPackage*`?

> I think one thing that might work is to generically bind `GObject*` - then we just do casts on both sides?

Hmm, `GPtrArray`s aren't `GOBject`s though, no?

One thing I did try was to use `glib::translate::PtrArray` (the wrapper type for `GPtrArray*`), but I couldn't get it work (though I don't remember clearly what exactly I tried and what error it was giving me).

Reply:
Oh heh right they're not GObjects...

Reply:
Wait yes it is, there's just *also* a C++ `libdnf::Package`.

Reply:
Ahh I thought you were referring to the `GPtrArray` itself. Hmm, so you're suggesting a e.g. `rust::Vec<GObject*>`? I *think* that'll hit the same issue of cxx.rs not supporting pointers/references as vector elements.

Reply:
WDYT about the tip commit here?